core/src/main/java/org/infinispan/CacheStreamable.java

```diff
@@ -0,0 +1,120 @@
+package org.infinispan;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.BaseStream;
+
+/**
+ * Interface that signifies which streams are able to be used with a {@link Cache}.  This method
+ * is useful to hold a reference to any of the types while still being able to invoke some methods.
+ */
+public interface CacheStreamable<T, S extends BaseStream<T, S>> extends BaseStream<T, S> {
```

Comment:
Yeah TBH I couldn't think of a good name.  Unfortunately `BaseCacheStream` doesn't fit in my opinion as it implies it is abstract to me.  Maybe `CacheStreamOperations` ? Like I said I am kinda at a loss for the name.